pyomo/gdp/disjunct.py

```diff
@@ -212,7 +212,7 @@ def __getstate__(self):
     def set_value(self, expr):
         for e in expr:
             # The user gave us a proper Disjunct block
-            if isinstance(e, _DisjunctData):
+            if e.type() == Disjunct:
```

Comment:
This should result in test failures.  For the case where the user provides implicit Disjuncts with
```
m.d = Disjunction(expr=[m.x >= 5, m.x <= 5])
``
the `e` will not be a (derived) Component and will not have a `.type()` method.